src/main/java/domain/sample/Sample.java

```diff
@@ -0,0 +1,233 @@
+package domain.sample;
+
+import domain.InvalidDomainException;
+import domain.sample.events.DataMadeAvailable;
+import domain.sample.events.FailedQualityControl;
+import domain.sample.events.LibraryPrepared;
+import domain.sample.events.MetadataRegistered;
+import domain.sample.events.PassedQualityControl;
+import domain.sample.events.SampleReceived;
+import domain.sample.events.SampleSequenced;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * <p>A sample in the context of sample-tracking.</p>
+ *
+ * @since 2.0.0
+ */
+public class Sample {
+
+  private final SampleCode sampleCode;
+
+  private final List<SampleEvent> events = new ArrayList<>();
+
+  private final CurrentState currentState;
+
+  private Sample(SampleCode sampleCode) {
+    this.sampleCode = sampleCode;
+    currentState = new CurrentState();
+  }
+
+  public static Sample create(SampleCode sampleCode) {
+    return new Sample(sampleCode);
+  }
+
+  public static Sample fromEvents(SampleEvent... events) {
```

Comment:
What will this method be used for?

Reply:
Instead of calling
```java
Sample sample = Sample.create(sampleCode);
// get events
events.forEach(sample.addEvent);
```
you could use
```java
//get events
Sample sample = Sample.fromEvents(events);
```